cmd/credentialUtil.go

```diff
@@ -69,6 +71,47 @@ func GetUserOAuthTokenManagerInstance() *common.UserOAuthTokenManager {
 	return currentUserOAuthTokenManager
 }
 
+/*
+ * GetInstanceOAuthTokenInfo returns OAuth token, obtained by auto-login,
+ * for current instance of AzCopy.
+ */
+func GetOAuthTokenManagerInstance() (*common.UserOAuthTokenManager, error) {
+	var err error
+	autoOAuth.Do(func() {
+		var lca loginCmdArgs
+		if glcm.GetEnvironmentVariable(common.EEnvironmentVariable.AutoLoginType()) == "" {
+			err = errors.New("no login type specified")
+			return
+		}
+		// Fill up lca
+		switch glcm.GetEnvironmentVariable(common.EEnvironmentVariable.AutoLoginType()) {
+		case "SPN":
+			lca.applicationID = glcm.GetEnvironmentVariable(common.EEnvironmentVariable.ApplicationID())
+			lca.certPath = glcm.GetEnvironmentVariable(common.EEnvironmentVariable.CertificatePassword())
+			lca.clientSecret = glcm.GetEnvironmentVariable(common.EEnvironmentVariable.ClientSecret())
```

Comment:
Oops! This isn't correct. CertificatePassword is not the path of the certificate. We'll need a new env var.

Note: CertificatePassword is a separate parameter, in case the user passes in a encrypted certificate.

Reply:
Try looking into the normal flow for a Service Principal login, it should help you understand what needs to happen here.

---
In reply to: [517100301](https://github.com/Azure/azure-storage-azcopy/pull/1167#discussion_r517100301) [](ancestors = 517100301)